### DISCUSSION TITLE
modem gsm

### DISCUSSION THREAD

**Discussion by tcpipchip** [zephyrproject-rtos/zephyr#38780] [Q&A]
Hi

I am compiling the GSM code to NRF52840, but i have a question

My UART0 is working (console), but where do i define the GPIO used to UART_1 (GSM) ? (TX and RXD)

i am getting on console

[00:00:00.827,758] <inf> sample_gsm_ppp: Board 'nrf52840_mdk' APN 'java.claro.com.br' UART 'UART_1' device (nil) (modem_gsm)

it can´t open UART_1

I want a UART (not USB or RTT) to  console and one UART to modem!

Thanks

**Comment by ycsin** [comment]
Not sure if I understood you, for the latest main branch you will need something like [this](https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/net/gsm_modem/boards/frdm_uart2_dts.overlay) in an `.overlay` in your application directory or inside your `someboard.dts` file.

TX/RX gpios are usually already defined in the `*.dtsi` file which the `someboard.dts` includes, if you are using an in-tree board.
If you are using a custom board it will be something like this:

```dts
&usart1 {
	pinctrl-0 = <&usart1_tx_pc4 &usart1_rx_pc5>;
	current-speed = <115200>;
	status = "okay";
	cellular: quectel_ec21 {
		compatible = "zephyr,gsm-ppp";
		label = "QUECTEL EC21";
		status = "okay";
	};
};
```

**Comment by tcpipchip** [comment]
I am sorry sorry! Tx and rx are working! I can get IP using generic modem on Lisa u200 and Sara g350. I have to know why I can’t get the gateway and mask. Boths return 0.0.0.0

**Comment by ycsin** [comment]
> I have to know why I can’t get the gateway and mask. Boths return 0.0.0.0

I have no idea what are these, @rerickson1 @jukkar any idea?

**Comment by rerickson1** [comment]
> > I have to know why I can’t get the gateway and mask. Boths return 0.0.0.0
> 
> I have no idea what are these, @rerickson1 @jukkar any idea?

I would check the AT command documentation for the two modems you are using to see if they support the command to retrieve the gateway and mask.
Also, make sure your modem's firmware is up to date.

**Comment by tcpipchip** [comment]
Good morning team!

I am back!
Let me explain about my tests, more details

I am using the zephyr GENERIC MODEM driver...i am testing on LISA U200, SARA G350 and SARA G450,.
There modem  works with ppp , because i tested on Windows Dialup, Raspberry ppp, OpenWrt ppp and finally on Nuttx Rtos.

In the Nuttx Rtos, i see that really not  back the  gateway and dns, then really looks normal, because i can PING to anywhwere, look
![image](https://user-images.githubusercontent.com/11488460/134521192-d5186d51-b182-4517-bbe0-811cef0f0681.png)

The only difference on ZEPHYR generic modem is that there is no Authentication (PAP) during PPP, and i need that with my SIM operator to works.

Maybe if the ZEPHYR generic modem has a PAP option will solve the problem! Ahh, i remember that the generic modem can make DNS during ping...

I hope that help you this comments
